rfc/kubernetes-ui-enhancements/kubernetes-ui-enhancements.md

```diff
@@ -0,0 +1,260 @@
+
+*The Job metadata panel as seen in the infrastructure UI*
+
+#### Proposal
+
+##### Display Job metadata on the infrastructure view
+
+The Job metadata panel serves as an effective means of surfacing the status and attributes of deployed jobs. However, it cannot be accessed directly. The metadata panel is not visible to users landing on the infrastructure route. Users must follow a link from a pipeline execution or  manipulate the URL by hand to find a given job. If the job was not deployed using Spinnaker or the execution history is unavailable, no link will exist to view the Job UI. Feedback from Spinnaker users, reveals that there is an expectation that the Job UI panel would appear in the infrastructure view. 
+
+The infrastructure view can be updated to display a list of jobs deployed to each cluster. Selecting a job from this list would then display the job metadata panel. This change would increase the visibility of deployed jobs and make it easier for users to monitor job resources across multiple clusters within Spinnaker.
+
+##### Cron job improvements
+
+Cron jobs have additional attributes that are crucial in understanding how and when a given job will execute. There is a set of attributes present in other Kubernetes tools that is not available in Spinnaker’s Cron Job metadata panel. This includes:
+
+- Execution schedule
+- Last scheduled execution time
+- Concurrency policy
+- Suspension status. 
+
+Surfacing these attributes in the panel would allow users to quickly understand the state of their cron jobs across clusters. They can be added as collapsible windows to the existing panel UI.
+
+### Editing Actions
+
+New objects being exposed as part of this effort will not support editing actions from the infrastructure view. [Discussions](https://docs.google.com/document/d/1db_yw1uru99Byvin4lQgm7aUZ9xMmvsARtEiiNUrmBo/edit#heading=h.g5ms48w4nz3d) with the Kubernetes SIG indicate a preference for not including edit actions in the infrastructure UI and highlighted concerns that it encourages deployment anti-patterns. 
+
+### Daemon Sets / Stateful Sets
+#### Current Functionality
+
+Daemon Sets / Stateful Sets are currently rendered in Spinnaker’s infrastructure view. They leverage a similar UI to Replica Sets and successfully convey all key object attributes. Given that the Daemon Set / Stateful Set UI has near parity with other tooling, I recommended leaving this functionality unchanged for this project.
+
+### Namespaces
+#### Current Functionality
+
+There is no dedicated rendering of Namespace metadata or associated Resource Quotas in Spinnaker’s UI. Namespace information is only provided as a string identifying where a deployed object resides. While it would be possible to add functionality for exposing associated Resource Quota objects, this feature would be primarily of use to cluster administrators. Refactoring the Spinnaker UI to surface Namespace resource allocation would be of limited use to application teams.
+
```

Comment:
Is there any discussion of displaying non-service/ingress load balancers?  I'm specifically referring to the various Istio and similar service-mesh type resources - are there any plans to be able to see the resources backing, for example, a VirtualService?

Reply:
See the `Custom CRDs` section. Istio and the like would all fall into this category.